### PR TITLE
WT-12409 Add bash common functions. Tidy up dist scripts.

### PR DESCRIPTION
**Pull request by ershov**
Summary of changes:
* Add `common_functions.sh`.
* Use it in every script in `dist/` to locate the repository and sources.
* Since `s_fast` only check files changed since fork, now it checks that it's being run for a proper repo.

### CODE REVIEW

File: dist/common_functions.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  fi
+}
+
+# https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html#index-BASH_005fARGV
+check_fast_mode_flag() { [[ "${BASH_ARGV[$(($BASH_ARGC - 1))]}" == "-F" ]] && check_fast_mode; }
+
+last_commit_from_dev() { python3 "$DIST_DIR"/common_functions.py last_commit_from_dev; }
+
+use_pygrep() {
+  [[ `uname -s` == Darwin ]] && EGREP="$DIST_DIR/pygrep.py -E" FGREP="$DIST_DIR/pygrep.py -F" || EGREP=egrep FGREP=fgrep
```
[the remark this answers is not in the document]
Comment (ajmorton) [reply dist/common_functions.sh]: No worries, happy to leave this as is 👍

File: dist/s_all [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 # Run standard scripts.
 
-t=__wt.$$
+. `dirname -- ${BASH_SOURCE[0]}`/common_functions.sh cd_dist
```
[the remark this answers is not in the document]
Comment (ajmorton) [reply dist/s_all:5]: I'm still 50/50 on this one but I don't want to hold up the PR.
@etienneptl I'll let you play tie breaker.

[the remark this answers is not in the document]
Comment (ershov) [reply dist/s_all]: Done.

**Review by ajmorton — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**